### PR TITLE
neutron: Put ML2 Cisco config files as snippets in neutron.conf.d

### PR DESCRIPTION
**Pull request by vuntz**
This enables people to override values in these files with snippets.

Followup to https://github.com/crowbar/crowbar-openstack/pull/1466

### CODE REVIEW

File: chef/cookbooks/neutron/recipes/cisco_apic_support.rb [hunk 1]
```diff
@@ -20,8 +20,22 @@
   node[:neutron][:platform][:cisco_apic_gbp_pkgs].each { |p| package p }
 end
 
+# Empty the config file to avoid confusion
+file "/etc/neutron/plugins/ml2/ml2_conf_cisco_apic.ini" do
```
Comment (skazi0) [inline chef/cookbooks/neutron/recipes/cisco_apic_support.rb]: same as with ml2_conf_cisco.ini

File: chef/cookbooks/neutron/recipes/cisco_support.rb [hunk 2]
```diff
@@ -13,7 +13,20 @@
   switches = neutron[:neutron][:cisco_switches].to_hash
 end
 
-template "/etc/neutron/plugins/ml2/ml2_conf_cisco.ini" do
+# Empty the config file to avoid confusion
+file "/etc/neutron/plugins/ml2/ml2_conf_cisco.ini" do
```
Comment (skazi0) [inline chef/cookbooks/neutron/recipes/cisco_support.rb]: This is/can be a symlink to a file in neutron-server.conf.d which you delete later. The content from here will be lost and there will be a dangling symlink.

Comment (vuntz) [reply chef/cookbooks/neutron/recipes/cisco_support.rb]: I actually wonder how `file` deals with symlinks... Let me play with that.

Comment (vuntz) [reply chef/cookbooks/neutron/recipes/cisco_support.rb]: @skazi0 indeed, not as I expected. Fixed.

Comment (skazi0) [reply chef/cookbooks/neutron/recipes/cisco_support.rb]: @vuntz yeah, I had the same problem in my PR around this. Newer chef versions have `force_unlink` option for exactly this.

**Pull request merged**